tutorials/crio-in-kind.md

```diff
@@ -0,0 +1,254 @@
++++ [0508 15:45:22] Starting docker build for image: kube-proxy-arm64
++++ [0508 15:45:22] Starting docker build for image: kubectl-arm64
++++ [0508 15:45:32] Deleting docker image registry.k8s.io/kubectl-arm64:v1.30.0
++++ [0508 15:45:32] Deleting docker image registry.k8s.io/kube-scheduler-arm64:v1.30.0
++++ [0508 15:45:35] Deleting docker image registry.k8s.io/kube-proxy-arm64:v1.30.0
++++ [0508 15:45:35] Deleting docker image registry.k8s.io/kube-controller-manager-arm64:v1.30.0
++++ [0508 15:45:40] Deleting docker image registry.k8s.io/kube-apiserver-arm64:v1.30.0
++++ [0508 15:45:40] Docker builds done
+Finished building Kubernetes
+Building node image ...
+Building in container: kind-build-1715175957-1495463435
+Image "kindest/node:latest" build completed.
+```
+
+Now let's build our image with CRI-O on top of `kindest/node:latest`:
+
+<!-- markdownlint-disable MD013 -->
+```dockerfile
+FROM kindest/node:latest
+
+ARG CRIO_VERSION
+ARG PROJECT_PATH=prerelease:/$CRIO_VERSION
+
+RUN echo "Installing Packages ..." \
+    && apt-get clean \
+    && apt-get update -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    software-properties-common vim gnupg \
+    && echo "Installing cri-o ..." \
+    && curl -fsSL https://pkgs.k8s.io/addons:/cri-o:/$PROJECT_PATH/deb/Release.key | gpg --dearmor -o /etc/apt/keyrings/cri-o-apt-keyring.gpg \
+    && echo "deb [signed-by=/etc/apt/keyrings/cri-o-apt-keyring.gpg] https://pkgs.k8s.io/addons:/cri-o:/$PROJECT_PATH/deb/ /" | tee /etc/apt/sources.list.d/cri-o.list \
+    && apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get --option=Dpkg::Options::=--force-confdef install -y cri-o \
+    && sed -i 's/containerd/crio/g' /etc/crictl.yaml \
+    && systemctl disable containerd \
+    && systemctl enable crio
+```
+<!-- markdownlint-enable MD013 -->
+
+<!-- markdownlint-disable MD013 -->
+Next let's build image with
+[`prerelease:v1.30`](https://github.com/cri-o/packaging/blob/main/README.md#prereleases) CRI-O version:
+
+```sh
+$ CRIO_VERSION=v1.30
+$ docker build --build-arg CRIO_VERSION=$CRIO_VERSION -t kindnode/crio:$CRIO_VERSION .
+# some output
+```
+<!-- markdownlint-enable MD013 -->
+
+With the built `node image,` we can create the kind cluster:
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      criSocket: unix:///var/run/crio/crio.sock
+  - |
+    kind: JoinConfiguration
+    nodeRegistration:
+      criSocket: unix:///var/run/crio/crio.sock
+- role: worker
+  kubeadmConfigPatches:
+  - |
+    kind: JoinConfiguration
+    nodeRegistration:
+      criSocket: unix:///var/run/crio/crio.sock
+```
+
+## Create kind cluster
+
+Let's create a cluster:
+
+<!-- markdownlint-disable MD013 -->
+```sh
+$ kind create cluster --image kindnode/crio:$CRIO_VERSION --config ./kind-crio.yaml
```

Comment:
Is it possible to use podman for running the cluster as an alternative to docker?

Reply:
AFAIK yes ... BUT generating the node and base images requires docker. Or used to require :) 

Probably better if someone confirmed this.

Reply:
Yeah, AFAIK, there is a lot of different options, and by leveraging `KIND_EXPERIMENTAL_PROVIDER` variable, you can create a cluster with different runtimes:

- podman
- nerdctl.lima
- finch
- nerdctl